src/backend/tests/blockbuster.py

```diff
@@ -0,0 +1,70 @@
+import asyncio
+import os
+import ssl
+import time
+from socket import socket as _socket
+
+
+class BlockingError(Exception): ...
+
+
+def _raise_if_blocking(func):
+    def wrapper(*args, **kwargs):
+        try:
+            asyncio.get_running_loop()
+        except RuntimeError:
+            return func(*args, **kwargs)
+        msg = f"Blocking call to {func.__module__}.{func.__qualname__}"
+        raise BlockingError(msg)
+
+    return wrapper
+
+
+def _wrap_os_blocking(func):
+    def os_op(fd, *args, **kwargs):
+        try:
+            asyncio.get_running_loop()
+        except RuntimeError:
+            return func(fd, *args, **kwargs)
+        if os.get_blocking(fd):
+            msg = f"Blocking call to {func.__module__}.{func.__qualname__}"
+            raise BlockingError(msg)
+        return func(fd, *args, **kwargs)
+
+    return os_op
+
+
+def _wrap_socket_blocking(func):
+    def socket_op(self, *args, **kwargs):
+        try:
+            asyncio.get_running_loop()
+        except RuntimeError:
+            return func(self, *args, **kwargs)
+        if self.getblocking():
+            msg = f"Blocking call to {func.__module__}.{func.__qualname__}"
+            raise BlockingError(msg)
+        return func(self, *args, **kwargs)
+
+    return socket_op
+
+
+def init():
+    time.sleep = _raise_if_blocking(time.sleep)
```

Comment:
One drawback of patching time.sleep is that debugging with breakpoints in test will raise the `BlockingError`.
So this behavior should probably be configurable.

Reply:
Maybe a fixture? I debuge a lot through tests

Reply:
Yes, me too.
Maybe there could be en env var `RAISE_ON_BLOCKING_TIME_SLEEP` that we set by default to `false`. And that we set to `true` in CI ?

Reply:
I like that option

Reply:
I found another way by inspecting the call stack.
I've checked with PyCharm and will also check with VSCode.

Reply:
It's OK with VSCode.